neo/Network/P2P/TaskManager.cs

```diff
@@ -184,17 +184,26 @@ private void RequestTasks(TaskSession session)
             }
             else if (Blockchain.Singleton.Height < session.Version.StartHeight)
             {
-                UInt256 hash = Blockchain.Singleton.CurrentBlockHash;
-                for (uint i = Blockchain.Singleton.Height + 1; i <= Blockchain.Singleton.HeaderHeight; i++)
+                uint iStart = Blockchain.Singleton.Height + 1;
+                UInt256 hashToStart = Blockchain.Singleton.GetBlockHash(iStart);
+                while(globalTasks.Contains(hashToStart) && iStart + 1 <= Blockchain.Singleton.HeaderHeight)
                 {
-                    hash = Blockchain.Singleton.GetBlockHash(i);
-                    if (!globalTasks.Contains(hash))
-                    {
-                        hash = Blockchain.Singleton.GetBlockHash(i - 1);
-                        break;
-                    }
+                    iStart++;
+                    hashToStart = Blockchain.Singleton.GetBlockHash(iStart);
                 }
-                session.RemoteNode.Tell(Message.Create("getblocks", GetBlocksPayload.Create(hash)));
+                if(iStart == Blockchain.Singleton.HeaderHeight)
+                {
+                    session.RemoteNode.Tell(Message.Create("getblocks", GetBlocksPayload.Create(hashToStart)));
+                    return;
+                 }
+                 uint iFinish = iStart + 1;
+                 UInt256 hashToFinish = Blockchain.Singleton.GetBlockHash(iFinish);
+                 while (!globalTasks.Contains(hashToFinish) && iFinish + 1 <= Blockchain.Singleton.HeaderHeight)
```

Comment:
If iFinish reaches Blockchain.Singleton.HeaderHeight it is probably better to just not include the hashToFinish and let nodes give back however many they have up to 500.

Reply:
Hi @jsolman,
It is done here, https://github.com/neo-project/neo/pull/486/commits/c31d8f1e06d8bd83c2792f1687ceeeceba8fe72b.

However, maybe we should change from 500 to a lower limit there, something like 10.

Reply:
But it has some impacts.

Currently, the mempool ask is cut by this limit:
* https://github.com/neo-project/neo/blob/c64748ecbac3baeb8045b16af0d518398a6ced24/neo/Network/P2P/Payloads/InvPayload.cs#L27-L35
* https://github.com/neo-project/neo/blob/c258cfc243abddd715442c2684dc29a537188b57/neo/Network/P2P/ProtocolHandler.cs#L265-L269
* https://github.com/neo-project/neo/blob/61d7b7a2398bf24aba996fa98ff88e05f34b79aa/neo/Consensus/ConsensusService.cs#L298-L302

Maybe a specific variable for blocks case should be set if it detected that it is not necessary to ask for a number such as 500 in the aforementioned case.

Reply:
@jsolman, with this last idea you commented about ask for `MaxHashesCount` I do not know if this change proposed in this PR becomes really worth.
It would only save bandwidth when the node is behind and has some blocks in the middle of the missed ones.